tutorials/animation/index.md

```diff
@@ -0,0 +1,874 @@
+
+* How to add basic animation to a widget using `addListener()` and
+  `setState()`.
+* Every time the Animation generates a new number, the `addListener()`
+  function calls `setState()`.
+* How to define an AnimatedController with the required `vsync` parameter.
+* Explains the "`..`" syntax in "`..addListener`", also known as Dart's
+  _cascade notation_.
+
+</div>
+
+So far you've learned how to generate a sequence of numbers over time.
+Nothing has been rendered to the screen.
+To render with an Animation&lt;&gt; object, store the Animation object as a
+member of your Widget, then use its value to decide how to draw.
+
+Consider the following application that draws the Flutter logo without
+animation:
+
+<!-- skip -->
+{% prettify dart %}
+import 'package:flutter/material.dart';
+
+class LogoApp extends StatefulWidget {
+  LogoAppState createState() => new LogoAppState();
+}
+
+class LogoAppState extends State<LogoApp> {
+  @override
+  Widget build(BuildContext context) {
+    return new Center(
+      child: new Container(
+        margin: new EdgeInsets.symmetric(vertical: 10.0),
+        height: 300.0,
+        width: 300.0,
+        child: new FlutterLogo(),
+      ),
+    );
+  }
+}
+
+void main() {
+  runApp(new LogoApp());
+}
+{% endprettify %}
+
+The following shows the same code modified to animate the
+logo to grow from nothing to full size. Note that when
+defining an AnimationController, you must pass in a `vsync` object.
+The presence of `vsync` prevents offscreen animations from consuming
+unnecessary resources. You can use your stateful object (in this case,
+LogoAppState) as the vsync by adding the SingleTickerProviderStateMixin
+to the class definition.
+
+The changes from the non-animated example are highlighted:
+
+<!-- skip -->
+{% prettify dart %}
+[[highlight]]import 'package:flutter/animation.dart';[[/highlight]]
+import 'package:flutter/material.dart';
+
+class LogoApp extends StatefulWidget {
+  LogoAppState createState() => new LogoAppState();
+}
+
+class LogoAppState extends State<LogoApp> [[highlight]]with SingleTickerProviderStateMixin[[/highlight]] {
+  [[highlight]]Animation<double> animation;[[/highlight]]
+
+  [[highlight]]initState() {[[/highlight]]
+    [[highlight]]super.initState();[[/highlight]]
+    [[highlight]]AnimationController controller = new AnimationController([[/highlight]]
+        [[highlight]]duration: const Duration(milliseconds: 2000), vsync: this);[[/highlight]]
+    [[highlight]]animation = new Tween(begin: 0.0, end: 300.0).animate(controller)[[/highlight]]
+      [[highlight]]..addListener(() {[[/highlight]]
+        [[highlight]]setState(() {[[/highlight]]
+          [[highlight]]// the state that has changed here is the animation object’s value[[/highlight]]
+        [[highlight]]});[[/highlight]]
+      [[highlight]]});[[/highlight]]
+    [[highlight]]controller.forward();[[/highlight]]
+  [[highlight]]}[[/highlight]]
+
```

Comment:
this demo leaks the animation controller. You need a dispose method where you call animation.dispose().

Reply:
Thanks! The source code is under Tao's repo. I've asked him to fix.